src/server/search/doc_accessors.cc

```diff
@@ -88,7 +89,11 @@ SearchDocData StringMapAccessor::Serialize(search::Schema schema) const {
 }
 
 string_view JsonAccessor::GetString(string_view active_field) const {
-  buf_ = json_->get_value_or<string>(active_field, string{});
+  error_code ec;
+  auto path = jsoncons::jsonpath::make_expression<JsonType>(active_field, ec);
+  DCHECK(!ec);
```

Comment:
Creating a json path for every read is wasteful, it should be created once upon validation and stored inside the field info (?)